src/main/kotlin/onboarding/Problem7.kt

```diff
@@ -1,9 +1,32 @@
 package onboarding
 
 fun solution7(
-    user: String,
-    friends: List<List<String>>,
-    visitors: List<String>
+        user: String,
+        friends: List<List<String>>,
+        visitors: List<String>
 ): List<String> {
-    TODO("프로그램 구현")
+    var friendList : ArrayList<String> = ArrayList<String>()
+    var scoreMap : MutableMap<String, Int> = mutableMapOf<String, Int>()
+
+    for(element in friends){
+        if(element[0] == user)
+            friendList.add(element[1])
+        else if(element[1] == user)
+            friendList.add(element[0])
+    }
+
+    for(element in friends){
+        if(!element.contains(user) && friendList.contains(element[0]))
+            scoreMap[element[1]] = scoreMap.getOrDefault(element[1], 0) + 10
+        else if(!element.contains(user) && friendList.contains(element[1]))
+            scoreMap[element[0]] = scoreMap.getOrDefault(element[0], 0) + 10
+    }
+
+    for(element in visitors)
+        if(!friendList.contains(element))
+            scoreMap[element] = scoreMap.getOrDefault(element, 0) + 1
+
+
+    return scoreMap.toList().sortedByDescending { it.second }.map{ it.first }.take(5)
```

Comment:
take 와 같이 뽑아야할 원소의 갯수가 정해져있는 리스트는 asSequence()를 사용하여 Lazy evaluation을 활용하시는 것도 좋아보입니다!
```suggestion
    return scoreMap
        .asSequence()
        .sortedByDescending { it.value }
        .map { it.key }
        .take(5)
        .toList()
```

현재 점수순으로만 정렬하는 것 같은데, `추천 점수가 같은 경우는 이름순으로 정렬한다.`라는 조건이 있어서 정렬을 한번더 진행하셔야 될 것 같아요